build-config/src/main/resources/checkstyle/checkstyle.xml

```diff
@@ -99,79 +110,68 @@
       <property name="max" value="300"/>
     </module>
     <module name="MethodName"/>
-    <module name="MethodParamPad">
-      <property name="severity" value="error"/>
+    <module name="MethodParamPad"/>
+    <module name="MissingDeprecated">
+      <property name="skipNoJavadoc" value="false" />
     </module>
-    <module name="MissingDeprecated"/>
     <module name="MissingOverride"/>
-    <module name="ModifierOrder">
-      <property name="severity" value="error"/>
-    </module>
+    <module name="MissingSwitchDefault"/>
+    <module name="ModifierOrder"/>
     <module name="MutableException">
-      <property name="severity" value="error"/>
       <message key="mutable.exception" value="Fields in an exception class must be final"/>
     </module>
-    <module name="NeedBraces">
-      <property name="severity" value="error"/>
-    </module>
+    <module name="NeedBraces"/>
     <module name="NoFinalizer"/>
-    <module name="NoWhitespaceAfter">
-      <property name="severity" value="error"/>
-    </module>
+    <module name="NoWhitespaceAfter"/>
     <module name="NoWhitespaceBefore">
       <property name="allowLineBreaks" value="true"/>
       <property name="tokens" value="SEMI,DOT,POST_DEC,POST_INC"/>
-      <property name="severity" value="error"/>
     </module>
     <module name="OneTopLevelClass"/>
     <module name="OperatorWrap">
       <property name="option" value="eol"/>
       <property name="tokens" value="ASSIGN, DIV_ASSIGN, PLUS_ASSIGN, MINUS_ASSIGN, STAR_ASSIGN, MOD_ASSIGN, SR_ASSIGN, BSR_ASSIGN, SL_ASSIGN, BXOR_ASSIGN, BOR_ASSIGN, BAND_ASSIGN"/>
     </module>
+    <module name="OverloadMethodsDeclarationOrder">
+      <!-- Checks that overload methods are grouped together. -->
+    </module>
     <module name="PackageAnnotation"/>
-    <module name="PackageName"/>
+    <module name="PackageName">
+      <!-- Package names should begin with the com.opengamma domain.-->
+      <property name="format" value="^com\.opengamma(\.[a-z][a-z]*)*$"/>
```

Comment:
We have some cases where this is not true eg. CME generated code, but perhaps the rule is still worth having

Reply:
Generally I'd say we should be adding exclusions here for any generated code (particularly if how that code get generated isn't configurable in a way that would allow us to make it adhere to the standards).